### PR TITLE
Many changes

### PR DESCRIPTION
**Pull request by BlueSlug**
- tidied up page visibility management
- initial work on story language input
- text-to-speech now reads title and author, with correct byline for the locale (SJRK-7)

### CODE REVIEW

**Comment by waharnum** [comment]
Writing some general comments for future work:

* We should extract the interface-related elements shared between the editor/viewer in `sjrk.storyTelling.story` into a separate base mix-in grade to be used by them  - basically, everything except the model.  This will keep the grades cleaner. 
* Until we can figure out the interaction between the language list and the "enter your own language" input, I'd prefer to leave out the language list drop down and just have it be a free text field.

File: src/js/storyTelling-story.js [hunk 1]
```diff
@@ -59,9 +65,40 @@ https://raw.githubusercontent.com/waharnum/sjrk-storyTelling/master/LICENSE.txt
...
         }
     });
 
+    // Generates full story text, including the title, author name and story content.
+    // title: the title of the story
+    // author: the author's name/moniker
+    // content: the textual content of the story
+    // bylineTemplate: the template string for the byline,
+    //      the value "%storyAuthor" will be replaced by the author's name
+    sjrk.storyTelling.story.getFullStoryText = function (title, author, content, bylineTemplate) {
```
Comment (waharnum) [inline src/js/storyTelling-story.js]: Could this function be simplified by making use of `fluid.stringTemplate` in places?

File: src/js/storyTelling-story.js [hunk 2]
```diff
@@ -59,9 +65,40 @@ https://raw.githubusercontent.com/waharnum/sjrk-storyTelling/master/LICENSE.txt
...
+
+        return (title ? title + " " : "") + byLine + (title || byLine ? ". " : "") + content;
+    };
+
+    // messagesString: JSON string of the collection of messages for the current locale, there should be
+    //      a "message_storyAuthorText" key. If not, it will return undefined
+    sjrk.storyTelling.story.getBylineTemplate = function (messagesString) {
+        if (messagesString) {
+            var messages = JSON.parse(messagesString);
+            return messages.message_storyAuthorText;
```
Comment (waharnum) [inline src/js/storyTelling-story.js]: We should avoid hard coding the expected key, and instead store the default of `message_storyAuthorText` on the component.

**Comment by BlueSlug** [comment]
I've refactored the story component into a new story.base component, which contains only the model, and a new story.ui component, which refers to the base component as well as the templatedComponentWithLocalization, so it is now aware of the resourceLoader and we can get rid of the two offending functions to build the story text. We'll need to do some further work on constructing the text "properly", but I feel it's a much cleaner setup now.

**Pull request merged**